crates/agent/src/controllers/activation.rs

```diff
@@ -115,23 +193,263 @@ pub async fn update_activation<C: ControlPlane>(
+                status.shard_status = Some(ShardStatusCheck {
+                    count: 0,
+                    first_ts: now,
+                    last_ts: now,
+                    status: ShardsStatus::Pending,
+                });
+                return Ok(Some(NextRun::from_duration(shard_health_check_interval(
+                    state,
+                    0,
+                    ShardsStatus::Pending,
+                ))));
+            }
         } else {
             tracing::debug!(restart_at = %rt, "waiting for backoff before restarting failed task shards")
         }
     }
 
-    if status.next_retry.is_some() {
+    // If we're waiting to restart failed shards, then we're done for now.
+    // Also return now if this is not a task with shards to monitor.
+    if status.next_retry.is_some() || !has_task_shards(state) {
         return Ok(status
             .next_retry
             .map(|rt| NextRun::after(rt).with_jitter_percent(0)));
-    } else if status.recent_failure_count > 0 {
-        return Ok(Some(NextRun::after_minutes(60)));
-    } else {
-        return Ok(None);
     }
+
+    // At this point we're finished with any activations that are needed, and
+    // it's time to ensure that task shards have actually started successfully.
+    let next_run = update_shard_health(status, control_plane, state).await?;
+    Ok(next_run)
 }
 
-fn get_next_retry_time(now: DateTime<Utc>, failures: &[ShardFailure]) -> Option<DateTime<Utc>> {
+async fn update_shard_health<C: ControlPlane>(
+    status: &mut ActivationStatus,
+    control_plane: &C,
+    state: &ControllerState,
+) -> anyhow::Result<Option<NextRun>> {
+    let (current_shard_status, count) = status
+        .shard_status
+        .as_ref()
+        .map(|s| (s.status, s.count))
+        .unwrap_or((ShardsStatus::Pending, 0));
+    let wait_interval = shard_health_check_interval(state, count, current_shard_status);
+    let Some(from) = status
+        .shard_status
+        .as_ref()
+        .map(|check| check.last_ts)
+        .or(status.last_activated_at)
+    else {
+        anyhow::bail!(
+            "internal controller error: attempted to health check a spec that was never activated"
+        );
+    };
+
+    let next = from + wait_interval;
+    let now = control_plane.current_time();
+    if now < next {
+        // We're still waiting for the next health check
+        return Ok(Some(NextRun::after(next)));
+    }
+
+    // Check the status of the shards
+    let Some(task_type) = state
+        .live_spec
+        .as_ref()
+        .and_then(|s| to_ops_task_type(s.catalog_type()))
+    else {
+        anyhow::bail!("internal controller error: attempted to health check a catalog test");
+    };
+
+    let list_response = control_plane
+        .list_task_shards(state.data_plane_id, task_type, state.catalog_name.clone())
+        .await
+        .context("listing task shards")?;
+
+    if list_response.status != proto_gazette::consumer::Status::Ok as i32 {
```

Comment:
the gazette client already checks status and maps into an error:
https://github.com/estuary/flow/blob/master/crates/gazette/src/shard/mod.rs#L60